js/components/SubmitModal.js

```diff
@@ -0,0 +1,130 @@
+var React = require("react");
+
+let Modal = require('react-bootstrap/lib').Modal;
+let Button = require('react-bootstrap/lib').Button;
+let FormGroup = require('react-bootstrap/lib').FormGroup;
+let FormControl = require('react-bootstrap/lib').FormControl;
+let HelpBlock = require('react-bootstrap/lib').HelpBlock;
+
+class SubmitModal extends React.Component {
+	constructor(props) {
+		super(props);
+		this.close = this.close.bind(this);
+		this.handleChange = this.handleChange.bind(this);
+		this.handleSubmit = this.handleSubmit.bind(this);
+		this.state = {
+			showModal: false,
+			url: '',
+			title: '',
+			text: '',
+			showError: false
+		};
+	}
+
+	componentDidMount() {
+		this.setState({ showModal : this.props.showModal });
+	}
+
+	componentWillReceiveProps(nextProps) {
+		if (this.state.showModal != nextProps.showModal) {
+			this.setState({ showModal : nextProps.showModal });
+		}
+	}
+
+	close() {
+		this.setState({ 
+			url: '',
```

Comment:
Content of function `close()` can be overwritten in different modals. Standardising is good, but too much work in this case. How about no :)